frontend/src/metabase-lib/lib/metadata/Segment.ts

```diff
@@ -1,47 +1,32 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck
 import Base from "./Base";
+import Database from "./Database";
+import Table from "./Table";
+import { SegmentFilter } from "metabase-types/types/Query";
+
 /**
  * @typedef { import("./metadata").FilterClause } FilterClause
  */
 
 /**
  * Wrapper class for a segment. Belongs to a {@link Database} and possibly a {@link Table}
  */
-
 export default class Segment extends Base {
+  id!: number;
+  name!: string;
+  description!: string;
```

Comment:
Unlike other data structures like `Table` and `Card`, we seem to require a defined `description` string on Segments and Metrics